testslide/patch_attribute.py

```diff
@@ -56,6 +60,7 @@ def patch_attribute(
     key = (id(target), attribute)
 
     if isinstance(target, testslide.StrictMock):
+        target.__dict__["_type_validation"] = type_validation
```

Comment:
I believe this will change this StrictMock instance constructor parameter on the fly:

- StrictMock is created with "type_validation=True".
- We do `patch_attribute(strict_mock, "att", type_validation=False)`
- StrictMock instance is now has no type validation.

What I would expect, is for `patch_attribute(strict_mock, "att", type_validation=False)` to only affect the attribute being patched. I suppose you added this here, because otherwise, despite `patch_attribute(strict_mock, "att", type_validation=False)`, StrictMock would still type check this attribute later.

I suppose a way out, would be for StirctMock to have some private attribute like `_attributes_to_skip_type_validation`, and have `patch_attribute` update it instead.

Makes sense?

Reply:
PS: Either way, adding a test case specifically to cover patch_attribute/StrictMock integration is a good idea :-D

Reply:
Good idea, will do that.